### DISCUSSION TITLE
How to change type of text input fields

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#1327] [Help]
I'd like to add text input fields for emails, urls and telephone numbers and set the `type` field in the input html to `email, `url` and `tel` respectively. I can't see a way to do this in avo, there are no field types for those and I've tried something like this but it doesn't work:

```
html: {
    edit: {
      input: {
        type: "tel"
      }
    }
  }
  ```

Is there another way?

**Comment by adrianthedev** [comment]
That's not a bad idea actually. Have a way of adding custom attributes to inputs. [This](https://github.com/avo-hq/avo/pull/923/files) is a bigger PR where we add some attributes to the fields.

The alternative is to create a custom field. That will render something like a text field by default and you could update that one.
